### PR TITLE
Fix SchedulerPredicates test after event reason rename

### LINKED ISSUE #12709: e2e scheduler_predicates tests are failing
The following test cases of scheduler_predicate test are failing on jenkins:
- SchedulerPredicates validates that NodeSelector is respected.
- SchedulerPredicates validates MaxPods limit number of pods that are allowed to run.
- SchedulerPredicates validates resource limits of pods that are allowed to run.

Please see:
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce/8078/

### PR DESCRIPTION
**Pull request by gmarek**
Fixes #12709

### CODE REVIEW

**Comment by gmarek** [comment]
cc @wojtek-t

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -177,7 +182,14 @@ var _ = Describe("SchedulerPredicates", func() {
 
 		pods, err := c.Pods(api.NamespaceAll).List(labels.Everything(), fields.Everything())
 		expectNoError(err)
-		currentlyRunningPods := len(pods.Items)
+		var currentlyRunningPods, currentlyDeadPods int
+		for _, pod := range pods.Items {
```
Comment (wojtek-t) [inline test/e2e/scheduler_predicates.go]: Can you please extract it to the method (very similar code is in 3 places here) - sth like\
func (pod api.PodList) runningAndDealPods (int, int)

**Comment by wojtek-t** [comment]
Generally looks good  - just one nit

File: test/e2e/scheduler_predicates.go (hunk 1 above)
Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: Done.

File: test/e2e/scheduler_predicates.go [hunk 2]
```diff
@@ -31,12 +31,24 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+// Returns a number of currently running and not running Pods.
+func getPodsNumbers(pods *api.PodList) (runningPods, deadPods int) {
```
Comment (wojtek-t) [inline test/e2e/scheduler_predicates.go]: deadPods is not the best name in my opinion (i.e. those are also pending pods). Maybe just "notRunningPods" ?

Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: Done.

**Comment by wojtek-t** [comment]
LGMT

**Comment by gmarek** [comment]
Jenkins failure is a flake.

**Comment by gmarek** [comment]
@k8s-bot ok to test

**Pull request merged**